src/System/Hapistrano.hs

```diff
@@ -276,36 +278,82 @@ initConfig getLine' = do
   configFilePath <- (FilePath.</> "hap.yml") <$> Directory.getCurrentDirectory
   alreadyExisting <- Directory.doesFileExist configFilePath
   when alreadyExisting $ do
-    T.hPutStrLn stderr "'hap.yml' already exists"
+    hPutStrLn stderr "'hap.yml' already exists"
     exitFailure
+
   putStrLn "Creating 'hap.yml'"
-  defaults <- defaultInitTemplateConfig
-  let prompt :: Read a => T.Text -> a -> IO a
-      prompt title d = do
-        T.putStrLn $ title <> "?: "
-        x <- getLine'
-        return $
-          if null x
-            then d
-            else read x
-      prompt' :: Read a => T.Text -> (InitTemplateConfig -> T.Text) -> (InitTemplateConfig -> a) -> IO a
-      prompt' title f fd = prompt (title <> " (default: " <> f defaults <> ")") (fd defaults)
-
-  let yesNo :: a -> a -> T.Text -> a
-      yesNo t f x = if x == "y" then t else f
-
-  config <-
-    InitTemplateConfig
-      <$> prompt' "repo" repo repo
-      <*> prompt' "revision" revision revision
-      <*> prompt' "host" host host
-      <*> prompt' "port" (T.pack . show . port) port
-      <*> return (buildScript defaults)
-      <*> fmap (yesNo (restartCommand defaults) Nothing) (prompt' "Include restart command" (const "Y/n") (const "y"))
+
+  config <- generateUserConfig defaultInitTemplateConfig
 
   Yaml.encodeFile configFilePath config
   putStrLn $ "Configuration written at " <> configFilePath
 
+  where
+    prompt :: Show a => String -> a -> MParser a -> IO a
+    prompt parameterName def parser = do
+      userInput <- prompt' (parameterName <> " (default: " <> show def <> ")")
+      if null userInput then
+        pure def
+      else do
+        let parsed = M.parse (parser <* M.eof) "" userInput
+        case parsed of
+          Left err -> do
+            hPutStrLn stderr (M.errorBundlePretty  err)
+            prompt parameterName def parser
+          Right res -> pure res
```

Comment:
@alexisbcc What do you think about something like this:
```suggestion
      when (null userInput)
        pure def

      either
          (\err -> hPutStrLn stderr (M.errorBundlePretty err) >> prompt parameterName def parser)
          pure
          (M.parse (parser <* M.eof) "" userInput)
```
?

Reply:
Looks better thanks